examples/encoding/plot_rerp_encoder.py

```diff
@@ -0,0 +1,67 @@
+"""
+========================================
+Regression on continuous event-related data (rER[P/F])
+========================================
+
+This demonstrates how rER[P/F]s - regressing the continuous data - is a
+generalisation of traditional averaging. If all preprocessing steps
+are the same, no overlap between epochs exists, and all predictors are binary,
+regression is virtually identical to traditional averaging.
+
+If overlap exists and/or predictors are continuous, traditional averaging
+is inapplicable, but regression can still estimate effects.
+
+rERPs are described in:
+
+    Smith, N. J., & Kutas, M. (2015). Regression-based estimation of ERP
+    waveforms: II. Non-linear effects, overlap correction, and practical
+    considerations. Psychophysiology, 52(2), 169-189.
+"""
+# Authors: Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import sample
+from mne.encoding.model import EventRelatedRegressor
+from mne.encoding.feature import clean_inputs
+
+# Load and preprocess data
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname, preload=True).pick_types(
+    meg='grad', stim=True, eeg=False).filter(1, None, method='iir')
+
+# Set up events
+events = mne.find_events(raw)
+event_id = {'Aud/L': 1, 'Aud/R': 2}
+tmin, tmax = -.1, .5
+
+# Regular epoching
+picks = mne.pick_types(raw.info, meg=True)
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax, reject=None,
+                    baseline=None, preload=True, verbose=False)
+
+# rERF
+rerp = EventRelatedRegressor(raw, events, est='cholesky', event_id=event_id,
+                             tmin=tmin, tmax=tmax,
+                             preproc_func_xy=clean_inputs)
```

Comment:
I don't like splitting up event related and continuous regressors. The underlying math is precisely the same. It's just that for event related ones, you can construct X from events and an event_id.
(In linear_regression_raw, this is handled by a preproc function that can take - simultaneously - both continuous and event related predictors, and outputs one predictor matrix. I think a helper to construct n length=n time points vector from an events matrix and n event_ids is better: clearer, and more powerful.)

Most importantly, beyond saving a class and some conceptual confusion, you want to be able to estimate joint event related and continuous models.

Or am I misunderstanding the API here?
Do you get what I'm saying?